feat(wallet) user can repeat a Send transaction from activity view

stefandunca opened the pull request:
### Closes #12122
    
Enable user action to repeat a Send transaction from the activity view (HistoryView) and details view (TransactionDetailView).
    
Extend AppMain send modal entry and SendModal API to allow for selecting all the required parameters for repeating a transaction.
    
Optimize the update of the start timestamp for the activity filter only when a user attempts to open the filter panel.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3a835718 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-statusq/job/PR-12237/1/) | 2023-09-24 08:12:34 | ~5 min | `tests/statusq` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-statusq/job/PR-12237/1//consoleText) |

stefandunca commented:
@Khushboo-dev-cpp I removed the new APIs from the send modal and moved them to `send.Helpers`. Also dropped the storybook changes. Please have another look.

Khushboo-dev-cpp reviewed (approved):
LGTM @stefandunca ! Awesome job

noeliaSD commented on ui/imports/shared/popups/send/SendModal.qml:
```diff
@@ -24,9 +24,14 @@ import "./views"
     property string preDefinedAmountToSend
+    // requires to have assigned an item from assets model
     property var preSelectedHolding
+    // token symbol
```
leftover??

stefandunca replied:
No, that should hold the token symbol in case of `preSelectedRecipientType: TabAddressSelectorView.Type.Address`. I learned this from @Khushboo-dev-cpp

noeliaSD commented on ui/imports/shared/popups/send/Helpers.qml:
```diff
@@ -0,0 +1,69 @@
+        }
+    }
+
+    function lookupAddressesForSendModal(senderAddress, recipientAddress, asset, isCollectible, amount) {
```
```suggestion
    function lookupAddressesForSendModal(senderAddress, recipientAddress, token, isCollectible, amount) {
```

noeliaSD replied:
Just a naming suggestion since in the UI context generally it is used `asset` to define `ERC20` tokens and `collectible` for `ERC721`, `token` as a general term.

stefandunca replied:
I changed it. Still, my understanding is the other way that asset is the general term for any kind of contract-based token or native currency that this variable can hold.

noeliaSD replied:
Maybe yes in general terms but in the UI context that's the differentiation we have right now! Thanks for the update!

noeliaSD commented on ui/imports/shared/popups/send/Helpers.qml:
```diff
@@ -0,0 +1,69 @@
+
+        if (isCollectible) {
+            req.preSelectedHoldingType = Constants.HoldingType.Collectible
+            req.preSelectedHolding = asset
```
```suggestion
            req.preSelectedHolding = token
```

noeliaSD commented on ui/imports/shared/popups/send/Helpers.qml:
```diff
@@ -0,0 +1,69 @@
+            req.preSelectedHolding = asset
+        } else {
+            req.preSelectedHoldingType = Constants.HoldingType.Asset
+            req.preSelectedHoldingID = asset
```
```suggestion
            req.preSelectedHolding = token
```

noeliaSD commented on ui/imports/shared/popups/send/Helpers.qml:
```diff
@@ -0,0 +1,69 @@
+
+        if (isCollectible) {
+            req.preSelectedHoldingType = Constants.HoldingType.Collectible
+            req.preSelectedHolding = asset
```
Is the expected `asset` the complete token `object` or and `id`? I think you need to review `lines 59` and/or `62`!!

stefandunca replied:
The asset is either a complete object representing an NFT or a symbol representing ERC20 or native currency.

stefandunca replied:
The comments were lost in the last refactoring, will update the function description to clarify this.

noeliaSD reviewed (commented):
Great job!
I've not tested the PR just added some minor comments in here!

stefandunca commented:
> Great job! I've not tested the PR just added some minor comments in here!

I addressed the comments, please have another look.
